packages/worker/src/worker-options.ts

```diff
@@ -420,6 +426,54 @@ export function addDefaultWorkerOptions(options: WorkerOptions): WorkerOptionsWi
   };
 }
 
+/**
+ * Inspect the execution environment and return information about available system resources,
+ * taking into account applicable constraints that can be discovered.
+ * At present, it accounts for CGroups v1 and v2 constraints on memory and swap.
+ */
+function inspectSystemResources(): { memory: number; heap: number } {
+  const resources = {
+    memory: os.totalmem(),
+    heap: v8.getHeapStatistics().heap_size_limit,
+  };
+
+  if (process.platform === 'linux') {
+    // v2 style cgroup
+    const cgroupv2MemoryMax = Number(tryReadFileSync('/sys/fs/cgroup/memory.max'));
+    if (!isNaN(cgroupv2MemoryMax)) resources.memory = Math.min(cgroupv2MemoryMax, resources.memory);
+
+    // v1 style cgroup
+    const cgroupv1MemoryMax = Number(tryReadFileSync('/sys/fs/cgroup/memory/memory.limit_in_bytes'));
```

Comment:
Let's avoid reading this file if we read the cgroups v2 file successsfully.

Reply:
Right... Also realized that the warning will show up once per worker. Fix and some refactoring coming in...